src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -4175,7 +4175,7 @@ void SmartScript::FillScript(SmartAIEventList e, WorldObject* obj, AreaTrigger c
         if (at)
             sLog->outDebug(LOG_FILTER_DATABASE_AI, "SmartScript: EventMap for AreaTrigger %u is empty but is using SmartScript.", at->entry);
 #endif
-        return;
+        return (void)at;
```

Comment:
Why? Just leave it as `return;`, don't cast it

Reply:
have a better idea. Post here

Reply:
Hmm, would be better to remove it here and add it at the beginning of the function with an appropriate comment, e.g.:
```cpp
void SmartScript::FillScript(SmartAIEventList e, WorldObject* obj, AreaTrigger const* at)
{
    (void)at; // ensure that the variable is referenced even if extra logs are disabled in order to pass compiler checks

    if (e.empty())
    {
```